ruma-events/src/membership.rs

```diff
@@ -0,0 +1,12 @@
+//! Membership types for the fedration API.
+
+use serde::{Deserialize, Serialize};
+
+use crate::room::member::MembershipState;
+
+/// MembershipEventContent
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MembershipEventContent {
```

Comment:
I don't think this should be a separate type from `MemberEventContent`. The federation endpoints don't expect any fields besides `membership` according to their docs, but I'm sure they're still supposed to store additional data so those endpoints should just use `Raw<MemberEventContent>` over `MembershipEventContent` in those endpoints.